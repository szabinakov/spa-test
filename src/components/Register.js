import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import axios from "axios";
import PropTypes from "prop-types";
import "../styles/Form.css";
import Alert from "./Alert.js";

export default function Register({ setUserState }) {
  const initialState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };
  const [alert, setAlert] = useState(initialState.alert);
  const [value, setValue] = useState();
  const history = useHistory();

  const handleChange = (event) => {
    setValue({
      ...value,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (value.password === value.confirmPassword) {
      console.log("Password Matches");
      axios
        .post("https://reqres.in/api/register", {
          email: value.email,
          password: value.password,
        })
        .then((response) => {
          setUserState({
            isUserLoggedIn: true,
          });
          history.push("/");
        })
        .catch(() => {
          setAlert({
            message: "Registration failed, please try again!",
            isSuccess: false,
          });
        });
    }
  };
  return (
    <>
      <Alert message={alert.message} success={alert.isSuccess} />
      <div className="formContainer">
        <h1>Please fill out the details below!</h1>
        <form className="form" onSubmit={handleSubmit}>
          <div>
            <input
              type="text"
              id="email"
              className="email"
              name="email"
              placeholder="Enter your email address!"
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <input
              type="password"
              id="password"
              className="password"
              name="password"
              placeholder="Enter your chosen password!"
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <input
              type="password"
              id="confirmPassword"
              className="confirmPassword"
              name="confirmPassword"
              placeholder="Confirm your password!"
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <button type="submit">Register</button>
          </div>
          <div>
            <Link to="/">
              <p>Already member? Login here!</p>
            </Link>
          </div>
        </form>
      </div>
    </>
  );
}

Register.propTypes = {
  setUserState: PropTypes.func,
};
