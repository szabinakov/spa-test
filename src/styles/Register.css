.registerFormContainer {
  margin-top: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.registerFormContainer > h3 {
  font-weight: lighter;
}
.registerForm {
  background-color: rgba(239, 71, 110);
  border-radius: 20px;
  box-shadow: 10px 10px 20px;
  padding: 10px;
  min-width: 260px;
}
.registerForm > div:first-child {
  margin-top: 0px;
}
.registerForm > div:last-child {
  margin-bottom: 0px;
}

.registerForm > div {
  display: flex;
  flex-direction: column;
  margin: 15px;
}

.registerForm > div > label {
  align-self: center;
  padding-bottom: 9px;
}

.registerForm > div > input {
  height: 20px;
  width: 80%;
  align-self: center;
  border-radius: 4px;
  box-shadow: 3px 3px 3px;
}

.registerForm > div > button {
  width: 90px;
  height: 25px;
  border-radius: 20px;
  align-self: center;
  text-transform: uppercase;
}

.registerForm > div > a > p {
  text-align: center;
  font-size: 10px;
}
