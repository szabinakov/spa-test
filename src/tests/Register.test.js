import { render } from "@testing-library/react";
import Register from "../components/Register.js";
import { BrowserRouter as Router } from "react-router-dom";

describe("Register", () => {
  const mocksetUserStateFunc = jest.fn();

  it("Matches the snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Register setUserState={mocksetUserStateFunc} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders every element", () => {
    const { getByTestId } = render(
      <Router>
        <Register setUserState={mocksetUserStateFunc} />
      </Router>
    );

    expect(getByTestId("form")).toHaveClass("form");
    expect(getByTestId("email")).toHaveClass("email");
    expect(getByTestId("password")).toHaveClass("password");
    expect(getByTestId("confirmPassword")).toHaveClass("confirmPassword");
    expect(getByTestId("buttonDiv")).toHaveClass("buttonDiv");
    expect(getByTestId("link")).toHaveClass("link");
  });
});
