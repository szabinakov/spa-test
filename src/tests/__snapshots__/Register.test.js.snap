// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="formContainer"
  >
    <h1>
      Please fill out the details below!
    </h1>
    <form
      class="form"
      data-testid="form"
    >
      <div>
        <input
          class="email"
          data-testid="email"
          id="email"
          name="email"
          placeholder="Enter your email address!"
          required=""
          type="text"
        />
      </div>
      <div>
        <input
          class="password"
          data-testid="password"
          id="password"
          name="password"
          placeholder="Enter your chosen password!"
          required=""
          type="password"
        />
      </div>
      <div>
        <input
          class="confirmPassword"
          data-testid="confirmPassword"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="Confirm your password!"
          required=""
          type="password"
        />
      </div>
      <div
        class="buttonDiv"
        data-testid="buttonDiv"
      >
        <button
          type="submit"
        >
          Register
        </button>
      </div>
      <div
        class="link"
        data-testid="link"
      >
        <a
          href="/"
        >
          <p>
            Already member? Login here!
          </p>
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
